src/native/managed/cdacreader/tests/MethodDescTests.cs

```diff
@@ -1,89 +1,137 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System;
+using System.Collections.Generic;
 using Microsoft.Diagnostics.DataContractReader.Contracts;
+using Microsoft.Diagnostics.DataContractReader.RuntimeTypeSystemHelpers;
 using Moq;
 using Xunit;
 
 namespace Microsoft.Diagnostics.DataContractReader.Tests;
 
 public class MethodDescTests
 {
-    private static void MethodDescHelper(MockTarget.Architecture arch, Action<MockDescriptors.MethodDescriptors> configure, Action<Target> testCase)
+    private static Target CreateTarget(MockDescriptors.MethodDescriptors methodDescBuilder)
     {
-        TargetTestHelpers targetTestHelpers = new(arch);
-
-        MockMemorySpace.Builder builder = new(targetTestHelpers);
-        MockDescriptors.RuntimeTypeSystem rtsBuilder = new(builder);
-        MockDescriptors.Loader loaderBuilder = new(builder);
-
-        var methodDescChunkAllocator = builder.CreateAllocator(start: 0x00000000_20002000, end: 0x00000000_20003000);
-        var methodDescBuilder = new MockDescriptors.MethodDescriptors(rtsBuilder, loaderBuilder)
-        {
-            MethodDescChunkAllocator = methodDescChunkAllocator,
-        };
-
-        configure?.Invoke(methodDescBuilder);
-
-        var target = new TestPlaceholderTarget(arch, builder.GetReadContext().ReadFromTarget, methodDescBuilder.Types, methodDescBuilder.Globals);
+        MockMemorySpace.Builder builder = methodDescBuilder.Builder;
+        var target = new TestPlaceholderTarget(builder.TargetTestHelpers.Arch, builder.GetReadContext().ReadFromTarget, methodDescBuilder.Types, methodDescBuilder.Globals);
         target.SetContracts(Mock.Of<ContractRegistry>(
             c => c.RuntimeTypeSystem == ((IContractFactory<IRuntimeTypeSystem>)new RuntimeTypeSystemFactory()).CreateContract(target, 1)
-                && c.Loader == ((IContractFactory<ILoader>)new LoaderFactory()).CreateContract(target, 1)));
-
-        testCase(target);
+                && c.Loader == ((IContractFactory<ILoader>)new LoaderFactory()).CreateContract(target, 1)
+                && c.PlatformMetadata == new Mock<Contracts.IPlatformMetadata>().Object));
+        return target;
     }
 
     [Theory]
     [ClassData(typeof(MockTarget.StdArch))]
     public void MethodDescGetMethodDescTokenOk(MockTarget.Architecture arch)
     {
-        TargetPointer testMethodDescAddress = default;
-        TargetPointer objectMethodTable = default;
+        TargetTestHelpers helpers = new(arch);
+        MockMemorySpace.Builder builder = new(helpers);
+        MockDescriptors.RuntimeTypeSystem rtsBuilder = new(builder);
+        MockDescriptors.Loader loaderBuilder = new(builder);
+        MockDescriptors.MethodDescriptors methodDescBuilder = new(rtsBuilder, loaderBuilder);
+
         const int MethodDefToken = 0x06 << 24;
         const ushort expectedRidRangeStart = 0x2000; // arbitrary (larger than  1<< TokenRemainderBitCount)
```

Comment:
Unrelated to your changes.

Can we assert this comment? In the sense `Debug.Assert(expectedRidRangeStart > (1<< TokenRemainderBitCount));`?